content/guides/getstarted/intro-to-ai.md

```diff
@@ -0,0 +1,544 @@
+[nvm](https://github.com/nvm-sh/nvm):
+
+```bash
+node --version
+nvm use 23
+```
+
+Then install the dependencies:
+
+```bash
+pnpm install --no-frozen-lockfile
+pnpm build
+```
+
+### Configure environment variables
+
+Create your environment file:
+
+```bash
+cp .env.example .env
+```
+
+What we are interested in for this guide is:
+
+```env
+TWITTER_USERNAME=               # Account username
+TWITTER_PASSWORD=               # Account password
+TWITTER_EMAIL=
+OPENAI_API_KEY=                 # Get this from: https://platform.openai.com/api-keys
+SOLANA_RPC_URL=https://api.devnet.solana.com # Lets work on devnet for now
+SOLANA_PRIVATE_KEY=             # Get this from: https://solana.com/wallet
+SOLANA_PUBLIC_KEY=
+SOLANA_CLUSTER=devnet # Default: devnet. Solana Cluster: 'devnet' | 'testnet' | 'mainnet-beta'
+SOLANA_ADMIN_PRIVATE_KEY=       # This wallet is used to verify NFTs (you can use the same as the SOLANA_PRIVATE_KEY for now)
+SOLANA_ADMIN_PUBLIC_KEY=        # This wallet is used to verify NFTs (you can use the same as the SOLANA_PUBLIC_KEY for now)
+```
+
+Make sure you have some
+[devnet sol](https://solana.com/de/developers/guides/getstarted/solana-token-airdrop-and-faucets)
+in your wallet before you start the agent.
+
+### Encode private key (if needed)
+
+If you have your private key in hex format you can base58 encode your private
+key using the following js snippet:
+
+```ts filename=encode58.ts
+const bs58 = require('bs58');
+console.log(bs58.encode(new Uint8Array([213,29,143,...])));
+```
+
+And run it using
+
+```bash
+npx tsx encode58.ts
+```
+
+### Create character configuration
+
+To create a new character I would recommend using the composer feature in Cursor
+and tell it what kind of character you want to create. Give it one of the other
+characters as input and describe what kind of character you want to create.
+
+```json filename=character.json
+{
+    "name": "Sol",
+    "clients": ["twitter"],
+    "modelProvider": "openai",
+    "settings": {
+        "voice": {
+            "model": "en_US-neural"
+        },
+        ...
+```
+
+The important part here are the clients and the modelProvider. In our case we
+want to use twitter as a client and openai as the model provider because this is
+what we also set our secrets for in the `.env` file.
+
+### Run the agent
+
+Now that the configuration is done we can run our character:
+
+```bash
+pnpm start --character="characters/sol.character.json"
+```
+
+This will start the agent and you can now interact with it on twitter. The
+character will automatically start posting tweets every few hours and when we
+interact with him it will be able to perform solana actions like minting NFTs
+for example.
+
+Here you can see the
+[character in action](https://x.com/solanadevhelper/status/1882222232656847143).
+
+You can also interact with the character locally using the following command:
+
+```bash
+pnpm start:client
+```
```

Comment:
I'm not able to get this working, it just stalls on disconnected

![CleanShot 2025-01-23 at 13 33 26@2x](https://github.com/user-attachments/assets/ef5b466d-b69c-4fc7-a34b-a1da0b172d19)

Reply:
Ill add trouble shooting section for this.